File: snippets/dart_snippets/lib/service_status.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<void> healthCheckStatus() async {
  // ANCHOR: health-check-status
  ServiceHealthCheckResponse healthCheck = await breezSDK.serviceHealthCheck(apiKey: "<api key>");
  print("Current service status is: ${healthCheck.status}");
  // ANCHOR_END: health-check-status
}

Future<void> reportPaymentFailure({required String paymentHash}) async {
  // ANCHOR: report-payment-failure
  ReportIssueRequest req = ReportIssueRequest.paymentFailure(
    data: ReportPaymentFailureDetails(
      paymentHash: paymentHash,
    ),
  );
  await breezSDK.reportIssue(req: req);
  // ANCHOR_END: report-payment-failure
}

File: snippets/swift/BreezSDKExamples/Sources/ServiceStatus.swift
//
//  ServiceStatus.swift
//
//
//  Created by dangeross on 27/11/2023.
//

import BreezSDK
import Foundation

func getServiceStatus() {
    // ANCHOR: health-check-status
    if let healthCheck = try? serviceHealthCheck(apiKey: "<api key>") {
        print("Current service status is: \(healthCheck.status)")
    }
    // ANCHOR_END: health-check-status
}

func reportPaymentFailure(sdk: BlockingBreezServices) {
    // ANCHOR: report-payment-failure
    let paymentHash = "..."

    try? sdk.reportIssue(
        req: ReportIssueRequest.paymentFailure(
            data: ReportPaymentFailureDetails(paymentHash: paymentHash)))
    // ANCHOR_END: report-payment-failure
}
